tardis/io/util.py

```diff
@@ -12,6 +13,34 @@
 import logging
 logger = logging.getLogger(__name__)
 
+float_pattern = r"""    # A numeric string consists of:
+    (?P<sign>[-+])?              # an optional sign, followed by either...
+
+    (
+
+        (?=\d|\.\d)              # ...a number (with at least one digit)
+
+        (?P<int>\d*)             # having a (possibly empty) integer part
+
+        (\.(?P<frac>\d*))?       # followed by an optional fractional part
+
+        (E(?P<exp>[-+]?\d+))?    # followed by an optional exponent, or...
+
+    |
+
+        Inf(inity)?              # ...an infinity, or...
+
+    |
+        NaN                      # NaN
```

Comment:
Regarding lower-case `nan` look at my previous replies.
Python supports and converts both +nan and -nan to just nan. This proves that there is no good reason to support signed nans, but also there is no good reason not to. :P